### PR TITLE
Restructured WordleGame.vue as there is now an asynchronous call to g…

### PR DESCRIPTION
**Pull request by htc36**
Restructured WordleGame.vue as there is now an asynchronous call to get solution from api endpoint, added some logic to get game solution object, fixed Wordle definition modal so it gets assets from online not local, added logic to not display game until everythings loaded.
Using localstorage currently to store game solution until we get more endpoints, the only place other than wordleGame.vue that gameSolution is used is wordDefintion.vue but there is potential to display the wordDefinition.vue everytime the user gets a correct maori word meaning we'd need logic anyway to get the words associated image url etc. 

In other words just keeping it simple for now till weve got the full picture of endpoints

### CODE REVIEW

**Comment by Mitri45** [comment]
Ok I assume it intermediate state, prob need to go deeper into best practices on api data handling in Vue 3 apps.
